blog/2019-12-05-exports-vs-export.md

```diff
@@ -0,0 +1,170 @@
+---
+title: exports、export 和 export default
+author: Robbie Han
+author_title: Front End Engineer @ Tradeshift
+author_url: https://github.com/USTC-Han
+author_image_url: https://robbie-pic.oss-cn-beijing.aliyuncs.com/IMG_4175.JPG?x-oss-process=style/compress
+tags: [exports, export, export default]
+---
+
+import Img from '../src/components/Img'
+
+## 前言
+在[exports 和 module.exports](https://thinkbucket.cn/blog/2019/12/03/exports-vs-module.exports)这篇博客中，我们详细解释了在CommonJS规范中`exports`和`module.exports`的区别。本篇博客将讨论CommonJS规范和ES6规范模块导出之间的关系。
+
+## export 和 export default
+ES6导出模块一般会使用`export`或`export default`，使用两者的语法在javaScript文档中的[ES6 Module](https://thinkbucket.cn/docs/javascript/6.modules/es6-module)已经做了详细的讲解，此处不再赘述。那么为什么已经有了`export`还要整个`export default`呢？`export`和`export default`之间的关系是什么呢？下面的内容也将通过这两个问题进行展开。
+
+### 为什么已经有了export还要整个export default呢？
+在ES6模块中，如果模块导出中使用`export var name = 'Robbie'`,那么在导入改模块的时候，导入的变量必须与之相对应，即`import {name} form Info`。而`export default name`在模块导入时的导入变量的命名可是任意的`import myName form Info`。对比两者来看使用`export default`可能具有更高的灵活性，也更便于不熟悉该库的开发者进行开发。
+
+### export和export default之间的关系是什么呢？
+我们知道浏览器是不支持ES6模块语法的，所以对于代码中的ES6模块一般是采用Babel转成CommonJS规范的代码。此时的代码虽然浏览器仍然不能执行，但是Node可以，因为webpack是基于Node构建的，所以我们可以通过webpack将代码打包成浏览器支持的ES5代码。
+
+<!--truncate-->
+
+<Img width="480" legend="图：ES6模块转浏览器可执行代码示意图" src="https://cosmos-x.oss-cn-hangzhou.aliyuncs.com/y8Gvm6.png" />
+
+为了探索ES6模块中`export`和`export default`这两个之间的关系，我们可以看下面这个例子
+
+```js
+  export var info = {age: 22, name: 'han'}
+  var email = 'SA17225105@gmail.com'
```

Comment:
personal email will be sensitive on Internet and suggest using fake email instead

Reply:
indent